code/game/machinery/computer/warrant.dm

```diff
@@ -0,0 +1,179 @@
+					background = "background-color:#990000;"
+					notice = "<br>**REPORT TO THE BRIG**"
+				if("Incarcerated")
+					background = "background-color:#CD6500;"
+				if("Paroled")
+					background = "background-color:#CD6500;"
+				if("Discharged")
+					background = "background-color:#006699;"
+				if("None")
+					background = "background-color:#4F7529;"
+				if("")
+					background = "''" //"'background-color:#FFFFFF;'"
+			dat += "<font size='4'><b>Warrant Data</b></font>"
+			dat += {"<table>
+			<tr><td>Name:</td><td>&nbsp;[current.fields["name"]]&nbsp;</td></tr>
+			<tr><td>ID:</td><td>&nbsp;[current.fields["id"]]&nbsp;</td></tr>
+			</table>"}
+			dat += {"Criminal Status:<br>
+			<div style='[background] padding: 3px; text-align: center;'>
+			<strong>[current.fields["criminal"]][notice]</strong>
+			</div>"}
+
+			dat += "<br><br>Citations:"
+
+			dat +={"<table style="text-align:center;" border="1" cellspacing="0" width="100%">
+			<tr>
+			<th>Crime</th>
+			<th>Fine</th>
+			<th>Author</th>
+			<th>Time Added</th>
+			<th>Amount Due</th>
+			<th>Make Payment</th>
+			</tr>"}
+			for(var/datum/data/crime/c in current.fields["citation"])
+				var/owed = c.fine - c.paid
+				dat += {"<tr><td>[c.crimeName]</td>
+				<td>$[c.fine]</td>
+				<td>[c.author]</td>
+				<td>[c.time]</td>"}
+				if(owed > 0)
+					dat += {"<td>$[owed]</td>
+					<td><A href='?src=[REF(src)];choice=Pay;field=citation_pay;cdataid=[c.dataId]'>\[Pay\]</A></td>"}
+				else
+					dat += "<td colspan='2'>All Paid Off</td>"
+				dat += "</tr>"
+			dat += "</table>"
+
+			dat += "<br>Minor Crimes:"
+			dat +={"<table style="text-align:center;" border="1" cellspacing="0" width="100%">
+			<tr>
+			<th>Crime</th>
+			<th>Details</th>
+			<th>Author</th>
+			<th>Time Added</th>
+			</tr>"}
+			for(var/datum/data/crime/c in current.fields["mi_crim"])
+				dat += {"<tr><td>[c.crimeName]</td>
+				<td>[c.crimeDetails]</td>
+				<td>[c.author]</td>
+				<td>[c.time]</td>
+				</tr>"}
+			dat += "</table>"
+
+			dat += "<br>Major Crimes:"
+			dat +={"<table style="text-align:center;" border="1" cellspacing="0" width="100%">
+			<tr>
+			<th>Crime</th>
+			<th>Details</th>
+			<th>Author</th>
+			<th>Time Added</th>
+			</tr>"}
+			for(var/datum/data/crime/c in current.fields["ma_crim"])
+				dat += {"<tr><td>[c.crimeName]</td>
+				<td>[c.crimeDetails]</td>
+				<td>[c.author]</td>
+				<td>[c.time]</td>
+				</tr>"}
+			dat += "</table>"
+		else
+			dat += {"<span>** No security record found for this ID **</span>"}
+	else
+		dat += {"<a href='?src=[REF(src)];choice=Login'>------------</a><hr>"}
+
+	var/datum/browser/popup = new(user, "warrant", "Security Warrant Console", 600, 400)
+	popup.set_content(dat.Join())
+	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
+	popup.open()
+
+/obj/machinery/computer/warrant/Topic(href, href_list)
+	if(..())
+		return
+
+	switch(href_list["choice"])
+		if("Login")
+			var/mob/M = usr
+			scan = M.get_idcard(TRUE)
```

Comment:
Do you actually need this scan variable ? You can just use current for the same purpose. Since if it exists you're logged in.

Reply:
Yea, scan is the current ID of the person logged in.